plenum/server/replica.py

```diff
@@ -435,7 +435,7 @@ def register_ledger(self, ledger_id):
         if ledger_id not in self.requestQueues:
             self.requestQueues[ledger_id] = OrderedSet()
         self._freshness_checker.register_ledger(ledger_id=ledger_id,
-                                                initial_time=self.get_current_time())
```

Comment:
`get_current_time` is used in unit tests to mock the time

Reply:
So, we need to 
1) Create a new field `get_time_for_3pc_batch` which can be overridden in tests similar to `get_current_time` and is `node.utc_epoch()` by default
2) Use `get_time_for_3pc_batch` when creating a PrePrepare, and when checking for freshness (`self._freshness_checker.check_freshness(self.get_time_for_3pc_batch())`)
3) Update all freshness unit tests to patch `get_time_for_3pc_batch` instead of `get_current_time`.